azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -116,6 +116,8 @@ def _init_cloud(self):
 
     def get_token(self):
         """Get a token silently from cache or authenticate if cached token is not found"""
+        if self.is_async:
+            raise KustoAsyncUsageError("Method get_token can't be called on an asynchronous client")

```

Comment:
Use format and pass the method name as an argument

Reply:
maybe: ..called **from** an.. ?

Reply:
WDYT of this way?